services/ecs.py

```diff
@@ -2,52 +2,226 @@
+    latest_task_definitions = {}
+
+    for task_definition in task_definitions:
+        if latest_task_definitions.get(task_definition.rsplit(":", 1)[0], 0) < int(task_definition.rsplit(":", 1)[1]):
+            latest_task_definitions[task_definition.rsplit(":", 1)[0]] = int(task_definition.rsplit(":", 1)[1])
+
+    for task_definition in latest_task_definitions.keys():
+        task_definition_arn = f"{task_definition}:{latest_task_definitions[task_definition]}"
+        task_definition = client.describe_task_definition(taskDefinition=task_definition_arn)["taskDefinition"]
+        containers = task_definition["containerDefinitions"]
+
+        for container in containers:
+            if container.get("readonlyRootFilesystem") == False:
+                non_compliant_resources.append(task_definition["taskDefinitionArn"])
+                break
+        else:
+            compliant_resources.append(task_definition["taskDefinitionArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ecs_container_insights_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    cluster_arns = client.list_clusters()["clusterArns"]
+
+    for cluster_arn in cluster_arns:
+        clusters = client.describe_clusters(clusters=[cluster_arn], include=["SETTINGS"])["clusters"]
+
+        for cluster in clusters:
+            settings = cluster["settings"]
+
+            for setting in settings:
+                if setting["name"] == "containerInsights" and setting["value"] == "enabled":
+                    compliant_resources.append(cluster_arn)
+                    break
+            else:
+                non_compliant_resources.append(cluster_arn)
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ecs_fargate_latest_platform_version():
+    compliant_resources = []
+    non_compliant_resources = []
+    cluster_arns = client.list_clusters()["clusterArns"]
+
+    for cluster_arn in cluster_arns:
+        services = client.list_services(cluster=cluster_arn, launchType="FARGATE")["serviceArns"]
+
+        for service in services:
+            service = client.describe_services(cluster=cluster_arn, services=[service])["services"][0]
+
+            if service["launchType"] == "FARGATE":
+                if service["platformVersion"] == "LATEST":
+                    compliant_resources.append(service["serviceArn"])
+                else:
+                    non_compliant_resources.append(service["serviceArn"])
+            else:
+                compliant_resources.append(service["serviceArn"])
```

Comment:
describe_services로 단일 service만 호출할 경우 성능이 크게 감소할 수 있습니다.
또한, FARGATE만을 조건으로 쿼리하였기 때문에 조건을 한번 더 확인하지 않아도 됩니다.
```suggestion
        service_arns = client.list_services(cluster=cluster_arn, launchType="FARGATE")["serviceArns"]
        services = client.describe_services(cluster=cluster_arn, services=service_arns)["services"]
        
        for service in services:
            if service["platformVersion"] == "LATEST":
                compliant_resources.append(service["serviceArn"])
            else:
                non_compliant_resources.append(service["serviceArn"])
```